cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/JmesPathOutputFilter.cs

```diff
@@ -3,25 +3,38 @@
 
 namespace Microsoft.Kiota.Cli.Commons.IO;
 
+/// <summary>
+/// An output filter that uses JMESPath queries to filter the output
+/// </summary>
 public class JmesPathOutputFilter : IOutputFilter
 {
     private readonly JmesPath jmesPath;
 
+    /// <summary>
+    /// Creates a new instance of JmesPathOutputFilter
+    /// </summary>
+    /// <param name="jmesPath">The JmesPath transformer instance.</param>
     public JmesPathOutputFilter(JmesPath jmesPath)
     {
         if (jmesPath is null) throw new ArgumentNullException(nameof(jmesPath), $"Parameter '{nameof(jmesPath)}' is required.");
         this.jmesPath = jmesPath;
     }
 
-    public Stream FilterOutput(Stream content, string query) {
+    /// <inheritdoc />
+    public async Task<Stream> FilterOutputAsync(Stream content, string query, CancellationToken cancellationToken = default) {
         if (string.IsNullOrEmpty(query)) return content;
+        cancellationToken.ThrowIfCancellationRequested();
```

Comment:
```suggestion
```

and remove other instances of this in this block

Reply:
adding throwIfCancellationRequested not only mixes concerns but also would require us to catch the task cancelled exceptions at some level or risk the user getting an exception.

Reply:
If you could clarify on the mixing of concerns point for context, I think it will help come up with a better solution. My thinking is that task status is a part of the function's domain so it should be communicated. Without the `ThrowIfCancellationRequested`, the function will complete with a [TaskStatus.RanToCompletion](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskstatus#system-threading-tasks-taskstatus-rantocompletion) status. In the case of the CLI, the cancellation token is cancelled when the user exits the program and I've set up the proper exception handling logic.

Reply:
reviewed on a call this is actually how it should be implemented. resolving